src/main/java/net/minestom/server/event/server/ServerListPingEvent.java

```diff
@@ -0,0 +1,201 @@
+    private final PlayerConnection connection;
+
+
+    public ServerListPingEvent(ResponseData responseData, PlayerConnection connection) {
+        this.responseData = responseData;
+        this.connection = connection;
+    }
+
+    /**
+     * ResponseData being returned.
+     *
+     * @return the response data being returned
+     */
+    public ResponseData getResponseData() {
+        return responseData;
+    }
+
+    /**
+     * HandshakeData of previous handshake packet
+     *
+     * equivalent to {@link #getConnection()#getHandshakeData()}
+     * @return
+     */
+    public HandshakeData getHandshakeData() {
+        return connection.getHandshakeData();
+    }
+
+    /**
+     * PlayerConnection of received packet.
+     *
+     * Note that the player has not joined the server at this time.
+     *
+     * @return the playerConnection.
+     */
+
+    public PlayerConnection getConnection() {
+        return connection;
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return cancelled;
+    }
+
+    /**
+     * Cancelling this event will cause you server to appear offline in the vanilla server list.
+     *
+     * @param cancel true if the event should be cancelled, false otherwise
+     */
+    @Override
+    public void setCancelled(boolean cancel) {
+        this.cancelled = cancel;
+    }
+
+    // Shortcut Methods
+    /**
+     * Sets the version name for the response.
+     *
+     * @param version The version name for the response data.
+     */
+    public void setVersion(String version) {
+        responseData.setVersion(version);
+    }
+
+    /**
+     * Sets the response protocol version.
+     *
+     * @param protocol The protocol version for the response data.
+     */
+    public void setProtocol(int protocol) {
+        responseData.setProtocol(protocol);
+    }
+
+    /**
+     * Sets the response maximum player count.
+     *
+     * @param maxPlayer The maximum player count for the response data.
+     */
+    public void setMaxPlayer(int maxPlayer) {
+        responseData.setMaxPlayer(maxPlayer);
+    }
+
+    /**
+     * Sets the response online count.
+     *
+     * @param online The online count for the response data.
+     */
+    public void setOnline(int online) {
+        responseData.setOnline(online);
+    }
+
+    /**
+     * Adds some players to the response.
+     *
+     * @param players the players
+     */
+    public void addPlayer(Iterable<Player> players) {
+        responseData.addPlayer(players);
+    }
+
+    /**
+     * Adds a player to the response.
+     *
+     * @param player the player
+     */
+    public void addPlayer(Player player) {
+        addPlayer(player.getUsername(), player.getUuid());
+    }
+
+    /**
+     * Adds a player to the response.
+     *
+     * @param name The name of the player.
+     * @param uuid The unique identifier of the player.
+     */
+    public void addPlayer(String name, UUID uuid) {
```

Comment:
Also @NotNull annotated - there's a few parameters in this class missing the annotations